pkg/devfile/adapters/common/command_test.go

```diff
@@ -421,6 +421,93 @@ func TestGetBuildCommand(t *testing.T) {
 
 }
 
+func TestGetDebugCommand(t *testing.T) {
+
+	command := "ls -la"
+	component := "alias1"
+	workDir := "/"
+	emptyString := ""
+
+	var emptyCommand common.DevfileCommand
+
+	tests := []struct {
+		name         string
+		commandName  string
+		execCommands []common.Exec
+		wantErr      bool
+	}{
+		{
+			name:        "Case: Default Debug Command",
+			commandName: emptyString,
+			execCommands: []common.Exec{
+				{
+					CommandLine: command,
+					Component:   component,
+					WorkingDir:  workDir,
+					Group: &versionsCommon.Group{
+						IsDefault: true,
+						Kind:      versionsCommon.DebugCommandGroupType,
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:        "Case: Custom Debug Command",
+			commandName: "customdebugcommand",
+			execCommands: []common.Exec{
+				{
+					Id:          "customdebugcommand",
+					CommandLine: command,
+					Component:   component,
+					WorkingDir:  workDir,
+					Group: &versionsCommon.Group{
+						IsDefault: false,
+						Kind:      versionsCommon.DebugCommandGroupType,
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:        "Case: Missing Debug Command",
+			commandName: "customcommand123",
+			execCommands: []common.Exec{
+				{
+					CommandLine: command,
+					Component:   component,
+					WorkingDir:  workDir,
+					Group: &versionsCommon.Group{
+						IsDefault: true,
+						Kind:      versionsCommon.BuildCommandGroupType,
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			devObj := devfileParser.DevfileObj{
+				Data: testingutil.TestDevfileData{
+					Components:   []common.DevfileComponent{testingutil.GetFakeComponent(component)},
+					ExecCommands: tt.execCommands,
+				},
+			}
+
+			command, err := GetDebugCommand(devObj.Data, tt.commandName)
+
+			if !tt.wantErr == (err != nil) {
```

Comment:
This condition is difficult to parse and somewhat obfuscates the intent. Also, since the second branch of the conditional is also testing for !tt.wantErr, I think it'd be better to have a top-level `if !tt.wantErr {` check and then split have an embedded `if/else` inside. Code would be clearer that way.

Reply:
Done